pkg/controllers/kube-storage-version-migrator.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright © 2021 MicroShift Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	klog "k8s.io/klog/v2"
+	migrationclient "sigs.k8s.io/kube-storage-version-migrator/pkg/clients/clientset"
+	"sigs.k8s.io/kube-storage-version-migrator/pkg/controller"
+)
+
+const (
+	kubeStorageVersionMigrator = "storage-version-migration-migrator"
+	kubeAPIQPS                 = float32(40.0)
+	kubeAPIBurst               = 1000
+)
+
+type KubeStorageVersionMigrator struct {
+	kubeconfig string
+	healthPath string
+	healthPort string
+}
+
+func NewKubeStorageVersionMigrator(cfg *config.Config) *KubeStorageVersionMigrator {
+	s := &KubeStorageVersionMigrator{}
+	s.kubeconfig = filepath.Join(cfg.KubeConfigRootAdminPath(), "kubeconfig")
+	s.healthPort = "2112"
+	s.healthPath = "storage-migrator-healthz"
+	return s
+}
+
+func (s *KubeStorageVersionMigrator) Name() string { return kubeStorageVersionMigrator }
+func (s *KubeStorageVersionMigrator) Dependencies() []string {
+	return []string{"kube-apiserver", "kubelet"}
+}
+
+func (s *KubeStorageVersionMigrator) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	errorChannel := make(chan error, 1)
+	defer close(stopped)
+
+	// run readiness check
+	go func() {
+		healthUrl := fmt.Sprintf("http://localhost:%s/%s", s.healthPort, s.healthPath)
+		healthcheckStatus := util.RetryInsecureGet(ctx, healthUrl)
+		if healthcheckStatus != 200 {
+			klog.ErrorS(fmt.Errorf("%s failed to start", s.Name()), "healthcheck failed", "name", s.Name())
+			errorChannel <- fmt.Errorf("%s healthcheck failed", s.Name())
+		}
+
+		klog.Infof("%s is ready", s.Name())
+		close(ready)
+	}()
+
+	go func() {
+		errorChannel <- s.runMigrator(ctx)
+	}()
+
+	return <-errorChannel
+}
+
+func (s *KubeStorageVersionMigrator) runMigrator(ctx context.Context) error {
+	go func() {
+		err := util.StartHealthCheck(s.healthPath, s.healthPort)
```

Comment:
Can you explain what's the plan for this endpoint?
Now it's static `ok`, will it contain some info about migration itself later? Does Trigger/Migrator need health endpoints of each other?

Reply:
Yeah, I couldn't decide just closing the ready checkpoint there, but I saw we did this pattern else where so I thought it might be good for now. But what do you think of just closing the ready channel here?

Reply:
We don't have to worry about this now